scrapy_dotpersistence.py

```diff
@@ -40,16 +40,14 @@ def __init__(self, crawler, bucket):
         self._load_data()
         crawler.signals.connect(self._store_data, signals.engine_stopped)
 
+    @property
+    def _s3path(self):
+        path = "/".join(
+            filter(None, [self._bucket, self._aws_username, self._projectid])
```

Comment:
Afais, each component is optional, which means that the `path` variable could easily be an empty string when running locally. Which leads to S3 urls like `"s3:///dot-scrapy/{spider}/"`, note 3 / characters in the beginning. Should we add an additional check to raise `NotConfigured` for the extension in such cases?

Reply:
The only required setting should be `ADDONS_S3_BUCKET`, so we get a path like `s3://s3_bucket/dot-scrapy/{spider}`. The PR was updated with this check.

AWS credentials should be mandatory too, but it would require changes that I consider not directly related to this PR. I'll create a new PR for all mandatory settings.

Reply:
Awesome, thanks!